web/viewer.css

```diff
@@ -309,12 +309,10 @@ html[dir='rtl'] #toolbarContainer, .findbar, .secondaryToolbar {
   left: 0;
   height: 100%;
   width: calc(100% + 150px);
-
   background: repeating-linear-gradient(135deg,
-                                        #bbb 0, #999 5px,
-                                        #999 45px, #ddd 55px,
-                                        #ddd 95px, #bbb 100px);
-
+                                        rgba(187,187,187,1) 0, rgba(153,153,153,1) 5px,
```

Comment:
*Nice idea, trying to consolidate the colours.*
 
The above line is just one example, but as far as I'm concerned it does highlight a slight problem that comes with disallowing hex values:
Looking at the hex values on the left, it's immediately obvious that the values are evenly spaced and symmetrical. With the RGBA format, not so much...

*Personally*, I find that the hexadecimal format can often be nicer to work with than the rgba one.

Hence my general question here, is if we could perhaps use the `#RRGGBB[AA]` or `#RGB[A]` formats instead?
I know that those formats have somewhat limited browser support (hi IE11/Edge), but there appears to be PostCSS plugins that could maybe handle the conversion for us: https://github.com/postcss/postcss/blob/master/docs/plugins.md

Reply:
I should mention that the idea for this commit series is to unify all color specifications into a single format while avoiding any CSS pre-/post-processing to keep the code and the build process simple.

I think the problem with the `#RRGGBB[AA]` format is not so much the limited browser support since that's fixable with post-processing, but I also find it harder to read. In the RGBA format, which seems most common for other projects too, the alpha channel is simply a value from 0 to 1 whereas in the `#RRGGBB[AA]` format that's less obvious because the alpha value is also in hexadecimal. Moreover, I don't see much difference in terms of this particular example, mainly because the values are aligned in a way that still allows you to see that there is symmetry. In short, I went for the RGBA format because of readability and because it's natively supported and often used, mainly with the idea of keeping it simple.

Initially I thought that switching to SCSS would be a good idea so that we could define the unique colors as variables. This would allow us to reduce the number of color specifications by expressing colors with opacities in terms of their root color, like done at https://stackoverflow.com/questions/21189495/setting-opacity-on-primary-color-in-sass. However, I quickly found that this would require conversions to the SCSS format and extra build steps for pre-processing, so I dismissed that idea. Then I thought that we could simply use native CSS variables, which only IE doesn't support, and that is fixable if we use PostCSS like you suggested. It would look something like https://stackoverflow.com/questions/40010597/how-do-i-apply-opacity-to-a-css-color-variable. Now that everything is in one format, the idea is that it's easier to extract the root colors into variables in a follow-up.

In conclusion, I'm now wondering what we best way forward is. Either we choose a simple and native solution, which is what this patch implements, or accept that we need to do pre-/post-processing in our build steps and use other techniques like `#RRGGBB[AA]` and CSS variables. Extracting colors into variables may be useful for us, but may also be overkill because we don't have *that* many colors, so I'm just not sure if we need all that complexity?!

What do you think about this? Should we prepare for CSS pre-/post-processing or should we try to avoid it for now?

Reply:
> [...] while avoiding any CSS pre-/post-processing to keep the code and the build process simple.

The reason that I even mentioned PostCSS plugins, is that we're already using Autoprefixer and PostCSS in the builds; see https://github.com/mozilla/pdf.js/blob/master/gulpfile.js

> [...] the alpha channel is simply a value from 0 to 1 whereas in the `#RRGGBB[AA]` format that's less obvious because the alpha value is also in hexadecimal.

I suppose that the part about the alpha channel behaviour indeed may look slightly confusing to users.

> Moreover, I don't see much difference in terms of this particular example, mainly because the values are aligned in a way that still allows you to see that there is symmetry.

We'll probably have to agree to disagree about the general readability/usability of hex vs rgb colour values, and I'll concede that this has a high degree of subjectivity attached :-)

> Then I thought that we could simply use native CSS variables, which only IE doesn't support, and that is fixable if we use PostCSS like you suggested.

Yes, there appears to be plugins which are able to inline CSS variables such that everything works in older browsers.
However, the idea of using CSS variables a lot more to improve both maintenance and customizability would seem like a nice feature. If such a re-factoring ever happens, having a build step that removes those variables thus seem like a bad idea. A much better option, assuming it would work of course, may be to try using something like https://github.com/jhildenbiddle/css-vars-ponyfill in `GENERIC` builds (to simply re-write CSS variables at runtime instead).

---

 - Generally though, if we're going to use the rgba format I think that the values ought to be spaced properly for improved readability (since we wouldn't allow "glued" values in e.g. JavaScript). Hence these values need to be formatted at `rgba(255, 255, 255, 1);` as far as I'm concerned.

 - Comparing the preview with `master`, there does appear to be colours which are now off-by-one. Using the Firefox devtools "Inspector", and looking at the "Computed" tab for e.g. the `#toolbarContainer` element there's a lot of small movement. Can this be fixed, please?

All in all, if the two points above are addressed I'd be happy to sign off on these changes :-)

Reply:
> Generally though, if we're going to use the rgba format I think that the values ought to be spaced properly for improved readability [...]

I agree with that; I'll change that.

> Comparing the preview with master, there does appear to be colours which are now off-by-one in all components [...]

Yes, it looks like my color converter tool got confused about the opacity part. I'll have to go over all occurrences again anyway to space them properly, so I'll also fix those occurrences.

Reply:
Both changes should be done now.